### ISSUE TITLE
Cant launch Fallout 4

### ISSUE THREAD

**Issue by t14462** [Tk-Glitch/PKGBUILDS#245]
It is regression, because my config same as for previous version.

Game just starting with black screen, and going to background.

This bug does not appearing in staging from binary repo

**Comment by skeevy420** [comment]
Got the exact same error with Skyrim & Hitman 2 last night when I did a staging 4.12.1.r3g5ed9 build of proton-tkg.  Black screen, but the audio works.

I noticed that there were some DXVK commits between then and now so I'm going to try a proton-tkg rebuild and see if that change anything.

**Comment by soredake** [comment]
Same for me with latest tkg.

**Comment by t14462** [comment]
Just... don't use:

# Bypass compositor in fullscreen mode - Reduces stuttering and improves performance - https://github.com/ValveSoftware/wine/commit/141ba5cf73029029a5a0bd2cdcfd5f9f9ab7ee7b
_FS_bypass_compositor="false"

# Proton Fullscreen patch - Requires staging & 3.16+ - Allows resolution changes for fullscreen games without changing desktop resolution - ! will enforce _FS_bypass_compositor="true" !
_proton_fs_hack="false"

This options incompatible with latest wine, it works somehow with some games, but I recommend to use KWin-lowlatency, and in-game tunable resolution like Mass Effect Andromeda and Dragon Age Inquisition

I think, Valve and CodeWeavers fix this problem in two weeks

**Comment by skeevy420** [comment]
I'll give that a shot and post back.  Probably in 30 or 40 minutes if I don't get sidetracked by life.

Finished building proton-tkg 4.12.1.r4g6530 with the updated DXVK and having the same issue.

**Comment by skeevy420** [comment]
Set both of those to false and I'm still having that issue.  Perhaps due to using the same staging commits and some issue when the installer overwrites the old version with the same name.

I'm gonna make sure it has a different name on a new build to rule that out.

**Comment by t14462** [comment]
Integrated DXVK builds as winelib, and it's a problem.

Use binary DXVK, or build it for yourself from here (tkg PKGBUILDS. install script included) - you need MinGW-automator for building, and it's here too.

Hmmm... my skyrim-se/DAI/Andromeda working with both options "true", but F4 cannot

**Comment by skeevy420** [comment]
Ahh, but I've never had any success using DXVK from winetricks...usually get the "No compatible graphics device found" error when using that on top of my proton builds.

Still worth a shot since I have everything installed.  I reckon I have less than 20 minutes left on my current proton-tkg build so I'll build DXVK then.

Would setting _use_dxvk="" & _dxvk_dxgi="" both to false be a good idea for this (and d9vk of course)?

**Comment by t14462** [comment]
In future (maybe) d9vk being merged to DXVK, but Joshua think it's not ready now.

But you can build D9VK from DXVK-tools and install it

**Comment by skeevy420** [comment]
Well, on the rebuild with the tweaked name the launcher didn't even launch.

Using your suggestions, setting the various dxvk settings to false/prebuilt to use my local build works with Hitman 2 & Skyrim SE.

Note for AMDGPU users -- I have the entire AMDGPU-Pro stack installed on my Manjaro system using the package from the AUR and neither of the games will launch with RADV but work just fine in my limited testing with AMDVLK-Pro.  I completely deleted the compatdata in-between the Vulkan tests to rule out any issues with the prefix.

I don't know if that is an AMDGPU-Pro issue or not, but I assume it is because I've never had that problem when only using AMDVLK from the Pro driver.

**Comment by t14462** [comment]
Same thing: Manjaro + AMDGPU + Mesa ACO

I think, AMD-Pro drivers is for... Pro. - Because with it you have good OpenCL for calculating and rendering your projects, but cool Mesa-things as Gallium Nine (DX9) inaccessible.

So i just waiting for official release of Mesa's RADV/ACO - because it opensource (you can update your kernel safely), and there is no big difference in performance.

**Comment by skeevy420** [comment]
I normally only use AMDVLK from the Pro package because it plays some games better than RADV, but earlier today I was trying to help a user over at Phoronix with an 5700XT issue so I installed the same amdgpu-pro package as them to see if it would work on my system...it does outside of this one issue that I've noticed.

**Comment by Tk-Glitch** [comment]
Without logs I can't help you guys.

**Comment by skeevy420** [comment]
My problem is solved.  I followed @t14462's advice.

Had to set:
```
_use_dxvk="prebuilt"
```
in proton-tkg.cfg and use the binareis provided by updxvk.  The other two weren't necessary for Skyrim SE or Hitman 2.  Didn't try anything else.

The Vulkan issues are unrelated to anything here/from your repo.

**Comment by Tk-Glitch** [comment]
What was it set to before?

**Comment by skeevy420** [comment]
true...ditto with dxgi.  Sometime after 4.11 r11 staging commit, my last good, working build until today, it just quit working, but due to all the staging changes I just figured I'd give it some time to settle down.

**Comment by Tk-Glitch** [comment]
I just removed DXVK/D9VK winelib support altogether. No more headache with these :)
Setting it to "true" will now simply act as if you used "release".

**Comment by soredake** [comment]
Still a black screen in f4 with latest dxvk-tkg and binary version.

**Comment by Tk-Glitch** [comment]
Still no log :frog:

**Comment by t14462** [comment]
> Still a black screen in f4 with latest dxvk-tkg and binary version.

In game shortcut: 
Insensitive: wine '/home/seth/.wine/drive_c/GAMES/Fallout 4/Fallout4.exe'

Working Path: /home/seth/.wine/drive_c/GAMES/Fallout 4/ (SENSITIVE)
So, maybe your working path incorrect (you typing "games" instead of "GAMES")

**Comment by soredake** [comment]
I use winesteam lutris runner.......

**Comment by soredake** [comment]
> Still no log frog

[last_build_config.zip](https://github.com/Tk-Glitch/PKGBUILDS/files/3392815/last_build_config.zip)
[f4_log.zip](https://github.com/Tk-Glitch/PKGBUILDS/files/3392829/f4_log.zip)

**Comment by Tk-Glitch** [comment]
Apparently other games will just give a black screen with recent wined3d changes. No idea how relevant this is for you @soredake 

That being said, I'm noticing something very wrong in your log : 
```
err:winediag:wined3d_dll_init Forcing sample count to 16. This may not be compatible with all applications.
```
This is apparently linked to an option lutris has added recently. This looks terrible.
Other than that it's pretty empty and I see nothing except for gamemodeauto (wtf?).

Also, regarding your build, you have an interesting combo.
- Mechwarrior Online fix -> terrible hack
- World of Final Fantasy hack -> terrible hack
- LoL 9.10+ fix -> terrible hack

I'd try disabling these to make sure your issue isn't related. I'll ping a friend to try reproducing the issue on my end.

**Comment by skeevy420** [comment]
The "gamemodeauto" is from using [Feral's GameMode](https://github.com/FeralInteractive/gamemode) and it can spam logs.  It's used to change the governor, nice level, etc when launching programs; shouldn't be the cause of any problems here.

**Comment by Tk-Glitch** [comment]
I am very well aware. The weird thing was for it to show up in a wine log. But I guess it's just that the logs are tinted by lutris and grabs whatever trash lying around.

**Comment by skeevy420** [comment]
Don't know about you, but it happens a lot when I run "gamemodeauto wine _anything_" to the point that I've considered just not using it.

**Comment by Tk-Glitch** [comment]
I don't see the point of using it.

**Comment by soredake** [comment]
Disabled these 3 patches and sample count and still black screen.

**Comment by Tk-Glitch** [comment]
![Screenshot_20190719_062258](https://user-images.githubusercontent.com/741977/61509298-14ee8f80-a9ee-11e9-9f40-34bd9df135fa.png)
I can't reproduce the issue. The game works just fine on my end. I'll need more details I guess.

**Comment by skeevy420** [comment]
> I don't see the point of using it.

Not having to remember to switch from schedutil to performance and back again when starting and stopping games, makepkg, or handbrake, I suppose.  I don't actually use it for anything else.

**Comment by Tk-Glitch** [comment]
This is a Bulldozer and Zen CPUs issue only though. Hopefully `amd_cpufreq` will fix that for you guys when 5.4 kernel drops.
